### PR TITLE
MacOS: Fix Xcode build

### PR DESCRIPTION
**Pull request by ann0see**
**Short description of changes**

Fixes building on macOS via Xcode. I'm not sure if that's the correct fix - but it seems to build fine. However, the app still crashes if the PlugIn folder of a working copy isn't copied into the Jamulus.app/Contents folder - but that's another bug.

Found by testing and not verified to actually apply the optimizations. 

There's still an issue that we need to copy the PlugIns folder
CHANGELOG: MacOS: Fix building Jamulus via Xcode

**Context: Fixes an issue?**
Fixes: https://github.com/jamulussoftware/jamulus/issues/1841 (at least partly)

**Does this change need documentation? What needs to be documented and how?**
No

**Status of this Pull Request**
Ready for review.

**What is missing until this pull request can be merged?**
Open another bug on the PlugIn bug.

## Checklist

-  [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/main/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
-  [x] I tested my code and it does what I want (not yet from this PR)
-  [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/main/CONTRIBUTING.md#source-code-consistency) 
-  [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
-  [x] I've filled all the content above

### CODE REVIEW

**Comment by ann0see** [comment]
Ok. I believe in general this PR is ready.

**Comment by ann0see** [comment]
@emlynmac could you please test if you can now build Jamulus on your Mac with Qt6?

**Comment by emlynmac** [comment]
> @emlynmac could you please test if you can now build Jamulus on your Mac with Qt6?

Sorry, I'm not able - none of my machines have the QT toolchain.

**Comment by ann0see** [comment]
Ok. Never mind. @danryu maybe?

File: Jamulus.pro [hunk 1]
```diff
@@ -1129,10 +1129,13 @@ contains(CONFIG, "opus_shared_lib") {
     DISTFILES += $$DISTFILES_OPUS
 
     contains(QT_ARCH, x86) | contains(QT_ARCH, x86_64) {
-        msvc {
+        msvc | macx-xcode {
             # According to opus/win32/config.h, "no special compiler
             # flags necessary" when using msvc.  It always supports
             # SSE intrinsics, but does not auto-vectorize.
+            # The macOS Xcode build also needs this - but optimizations
```
[the remark this answers is not in the document]
Comment (pljones) [reply Jamulus.pro:1136]: Yes - ideally, as with the msvc, quoting where that information comes from.

[the remark this answers is not in the document]
Comment (ann0see) [reply Jamulus.pro:1136]: I got to that by plain testing (trial and error) unfortunately.

[the remark this answers is not in the document]
Comment (pljones) [reply Jamulus.pro:1136]: Ah, fair enough.  Maybe add "(the Jamulus team has discovered)" or more individual if you prefer, just to indicate that?

[the remark this answers is not in the document]
Comment (ann0see) [reply Jamulus.pro]: Done.

**Review by pljones — APPROVED** [review]

**Pull request merged**